src/Arcus.Messaging.Pumps.Abstractions/MessagePump.cs

```diff
@@ -168,6 +182,26 @@ protected void NotifyResumeRetrievingMessages()
             Logger.LogDebug("Circuit breaker caused message pump '{JobId}' to transition back from '{CurrentState}' to a 'Closed' state, retrieving messages is resumed", JobId, CircuitState);
 
             CircuitState = MessagePumpCircuitState.Closed;
+
+            NotifyCircuitBreakerEventHandlers();
+        }
+
+        private void NotifyCircuitBreakerEventHandlers()
+        {
+            ICircuitBreakerEventHandler[] eventHandlers = GetEventHandlersForPump();
+
+            foreach (var handler in eventHandlers)
+            {
+                Task.Run(() => handler.OnTransition(CircuitState));
```

Comment:
I was wondering if this is a good way to trigger the eventhandlers in a fire-and-forget manner, and I came accross this interesting article:
https://michaelceber.medium.com/net-core-web-api-how-to-correctly-fire-and-forget-database-commands-from-controller-end-point-cad7a879630

The solution that is proposed there, offers the ability for the event-subscribers to access services that are registered in the DI container in a safe way.  Might be something to consider ?

Reply:
The blog talks mostly about moving the `Task.Run(...)` to an seperate service to not mix/match it with app code -> agree with that; but functionally, it is the same, though? Not sure why that would be an improvement as we're not showing any of this code to the users.

Reply:
That can also be done now, right? There is an overload for the event handler that lets you either regiser the type or provide an implementation factory function with the service-provider.

Reply:
And how can the service-provider be used within the eventhandler ?  That's not entirely clear to me atm.

Reply:
The event handlers are registered via the `IServiceCollection`. So, `.WithCircuitBreakerStateChangedEventHandler<T>(…)` has two overloads:
* One without additional arguments;
* One with an implementation factory function: `Func<IServiceProvider, TEventHandler>`.

When you register such an event handler, you have direct access to the service-provider.

```csharp
services.AddServiceBusMessagePump(…)
        .WithCircuitBreakerStateChangedEventHandler((IServiceProvider provider) =>
        {
            return new MyCircuitBreakerEventHandler(provider, …);
        });
```

Reply:
I would think that this is common knowledge as we follow the same practice as with Microsoft-developed services, which we follow for all our message handlers as well. Think this is even .NET Core library development best practices.

I would think that it would be too specific for feature docs? Especially since most of the time you don't need to access the service provider directly and can just inject?